src/supervisor-api.ts

```diff
@@ -0,0 +1,196 @@
+import * as express from 'express';
+import { Server } from 'http';
+import * as _ from 'lodash';
+import * as morgan from 'morgan';
+
+import Config from './config';
+import { EventTracker } from './event-tracker';
+import blink = require('./lib/blink');
+import * as iptables from './lib/iptables';
+import { checkTruthy } from './lib/validation';
+
+function getKeyFromReq(req: express.Request): string | null {
+	let headerKey: string | null = null;
+	const queryKey = req.query.apikey;
```

Comment:
You can add a
```ts
if (queryKey != null) {
	return queryKey
}
```
here to early exit since the `queryKey` takes priority, and then
```ts
if (!maybeHeaderKey) {
	return
}